service/discovery/openstack/account_discover.go

```diff
@@ -26,24 +26,63 @@
 package openstack
 
 import (
-	"clouditor.io/clouditor/v2/api/ontology"
+	"fmt"
 
+	"clouditor.io/clouditor/v2/api/discovery"
+	"clouditor.io/clouditor/v2/api/ontology"
 	"github.com/gophercloud/gophercloud/v2/openstack/identity/v3/domains"
 	"github.com/gophercloud/gophercloud/v2/openstack/identity/v3/projects"
 )
 
+// discoverDomains discovers domains.
+func (d *openstackDiscovery) discoverDomains() (list []ontology.IsResource, err error) {
+	var opts domains.ListOptsBuilder = &domains.ListOpts{}
+	list, err = genericList(d, d.identityClient, domains.List, d.handleDomain, domains.ExtractDomains, opts)
+
+	if err != nil {
+		// if we cannot retrieve the domain information by calling the API or from the environment variables, we will add the information manually if we already got the domain ID and/or domain name
+		log.Debugf("could not discover domains due to insufficient permissions, but we can proceed with less domain information: %v", err)
```

Comment:
Would start the sentence with capital letter "C" because it is directly printed and not forwarded to an error chain